nemo_skills/pipeline/utils.py

```diff
@@ -758,6 +848,52 @@ def get_cluster_config(cluster=None, config_dir=None):
     return cluster_config
 
 
+def update_ssh_tunnel_config(cluster_config: dict):
+    """
+    Update the ssh tunnel configuration in the cluster config to resolve job dir and username.
+    uses the `user` information to populate `job_dir` if in config.
+
+    Args:
+        cluster_config: dict: The cluster configuration dictionary
+
+    Returns:
+        dict: The updated cluster configuration dictionary
+    """
+    if 'ssh_tunnel' not in cluster_config:
+        return cluster_config
+
+    if 'user' in cluster_config['ssh_tunnel']:
+        # Resolve `user` from env if not provided
+        if cluster_config['ssh_tunnel']['user'] is None:
+            cluster_config['ssh_tunnel']['user'] = os.environ['USER']
+            LOG.info(f"Resolved `user` to `{cluster_config['ssh_tunnel']['user']}`")
+
+        elif isinstance(cluster_config['ssh_tunnel']['user'], str) and '$' in cluster_config['ssh_tunnel']['user']:
+            cluster_config['ssh_tunnel']['user'] = os.path.expandvars(cluster_config['ssh_tunnel']['user'])
+            LOG.info(f"Resolved `user` to `{cluster_config['ssh_tunnel']['user']}`")
+
+    if 'job_dir' in cluster_config['ssh_tunnel']:
+        # Resolve `job_dir` from env if not provided
+        cluster_config['ssh_tunnel']['job_dir'] = os.path.expandvars(cluster_config['ssh_tunnel']['job_dir'])
+        LOG.info(f"Resolved `job_dir` to `{cluster_config['ssh_tunnel']['job_dir']}`")
+
+    if 'identity' in cluster_config['ssh_tunnel']:
+        # Resolve `identity` from env if not provided
+        cluster_config['ssh_tunnel']['identity'] = os.path.expandvars(cluster_config['ssh_tunnel']['identity'])
+        LOG.info(f"Resolved `identity` to `{cluster_config['ssh_tunnel']['identity']}`")
+
+        if "$" in cluster_config['ssh_tunnel']['identity']:
+            raise ValueError(
+                f"Identity file path could not be loaded from env variable `{cluster_config['ssh_tunnel']['identity']}`"
+            )
+
+    # Add mount container list if not present
```

Comment:
feels out of place in this function as from the name it only supposed to update ssh section